src/google/protobuf/util/json_util.cc

```diff
@@ -105,13 +105,17 @@ util::Status BinaryToJsonString(TypeResolver* resolver,
 namespace {
 class StatusErrorListener : public converter::ErrorListener {
  public:
-  StatusErrorListener() : status_(util::Status::OK) {}
+  StatusErrorListener(const JsonParseOptions& options)
+      : options_(options), status_(util::Status::OK) {}
   virtual ~StatusErrorListener() {}
 
   util::Status GetStatus() { return status_; }
 
   virtual void InvalidName(const converter::LocationTrackerInterface& loc,
                            StringPiece unknown_name, StringPiece message) {
+    if (options_.ignore_unknown_fields) {
```

Comment:
This seems the wrong place to ignore unknown fields. When options.ignore_unknown_fields is enabled, this InvalidName() method shouldn't be called at all. Could you change the JSON parsing code to not call InvalidName?